File: Others/100問/1_全列挙/2_abc106_B.cpp
#include <bits/stdc++.h>
#include <cmath>
using namespace std;

#define ll long long
#define vec vector<int>
#define Graph vector<vector<int>>
#define rep(i, n) for (int i = 0; i < (int)(n); i++)
#define krep(i, k, n) for (int i = k; i < (int)(n); i++)

#define all(v) v.begin(), v.end()

/*
%d : int
%.5f : 下5桁
*/ 

int main() {
  // N,M
    int N;
    cin >> N;
    int ans = 0;
    krep(i,1,N+1){
        int cnt=0;
        krep(j,1,i+1){
            if(i%j==0){
                cnt += 1;
            }
        }
        if(cnt==8 && i%2==1){
            ans += 1;
        }
    }
    cout << ans << endl;
}

File: Others/100問/0_bk/2_abc106_B.py
N = int(input()) 

ans = 0
ls = [3,5,7,11,13]
# case 1
for i in ls:
    for j in ls:
        for k in ls:
            if i*j*k <= N and i<j and j<k:
                ans += 1
# case 2
for i in ls:
    for j in ls:
        if i*i*i*j <= N and i<j and j<k:
            ans += 1
            
        elif i*j*j*j < N and i<j and j<k:
            ans += 1
            
        else:
            pass
                
print(ans)
            